packages/itwin/tree-widget/src/components/trees/models-tree/ModelsTreeDefinition.ts

```diff
@@ -676,48 +652,22 @@ function createGeometricElementInstanceKeyPaths(
     const ecsql = `
       SELECT mce.ModelId, mce.Path, mce.GroupingNodeIndex
       FROM ModelsCategoriesElementsHierarchy mce
-      WHERE mce.ParentId IS NULL AND mce.GroupingNodeIndex = -1
-
-      UNION ALL
-
-      SELECT
-        ClosestGroupingNodeGroupedInstance.ModelId,
-        ClosestGroupingNodeGroupedInstance.Path,
-        ClosestGroupingNodeGroupedInstance.GroupingNodeIndex
-      FROM (
-        SELECT
-          mce.ModelId as ModelId,
-          mce.Path as Path,
-          mce.GroupingNodeIndex as GroupingNodeIndex,
-          ROW_NUMBER() OVER (PARTITION BY mce.GroupingNodeIndex ORDER BY mce.Depth) as RowNumber
-        FROM ModelsCategoriesElementsHierarchy mce
-        WHERE mce.ParentId IS NULL AND mce.GroupingNodeIndex <> -1
-      ) as ClosestGroupingNodeGroupedInstance
-      WHERE ClosestGroupingNodeGroupedInstance.RowNumber = 1
+      WHERE mce.ParentId IS NULL
     `;
     const createIdBinding = (id: Id64String) => ({ type: "id" as const, value: id });
-    const createCategoryElementsBindings = ({ ids, modelIds }: { ids: Id64String[]; modelIds: Id64String[] }) => [
-      ...ids.map(createIdBinding),
-      ...modelIds.map(createIdBinding),
-    ];
     return imodelAccess.createQueryReader(
       {
         ctes,
         ecsql,
-        bindings: [
-          ...elementIds.map(createIdBinding),
-          ...groupInfos.flatMap((info) =>
-            info.parent.type === "element" ? info.parent.ids.map(createIdBinding) : createCategoryElementsBindings(info.parent),
-          ),
-        ],
+        bindings: [...elementIds.map(createIdBinding), ...groupingNodes.map((info) => createIdBinding(info.groupedInstanceKeys[0].id))],
```

Comment:
did you try focusing multiple elements under a grouping node? are all of them loaded with the change?

Reply:
Fixed this by creating paths from all grouped elements